modules/services/workspace-service/src/main/java/org/ballerinalang/composer/service/workspace/langserver/LangServerManager.java

```diff
@@ -511,42 +525,42 @@ private void getBuiltInPackages(Message message) {
-            Position position = textDocumentPositionParams.getPosition();
-            ArrayList<CompletionItem> completionItems = new ArrayList<>();
-
-            BFile bFile = new BFile();
-            bFile.setContent(textContent);
-            bFile.setFilePath("/temp");
-            bFile.setFileName("temp.bal");
-            bFile.setPackageName(".");
-
-            AutoCompleteSuggester autoCompleteSuggester = new AutoCompleteSuggesterImpl();
-            CapturePossibleTokenStrategy capturePossibleTokenStrategy = new CapturePossibleTokenStrategy(position);
-            try {
-                ArrayList symbols = new ArrayList<>();
-                CompletionItemAccumulator completionItemAccumulator = new CompletionItemAccumulator(symbols, position);
-                BallerinaFile ballerinaFile =
-                        autoCompleteSuggester.getBallerinaFile(bFile, position, capturePossibleTokenStrategy);
-                capturePossibleTokenStrategy.getSuggestionsFilterDataModel().setBallerinaFile(ballerinaFile);
-                ballerinaFile.accept(completionItemAccumulator);
-
-                SuggestionsFilter suggestionsFilter = new SuggestionsFilter();
-                SuggestionsFilterDataModel dm = capturePossibleTokenStrategy.getSuggestionsFilterDataModel();
-                dm.setClosestScope(completionItemAccumulator.getClosestScope());
-                // set all the packages associated with the runtime. "this.getPackages()" might return null as process
-                // of loading packages is running in a separate thread. See initBackgroundJobs() method.
-                dm.setPackages(this.getPackages());
-
-                completionItems = suggestionsFilter.getCompletionItems(dm, symbols);
-            } catch (IOException e) {
-                this.sendErrorResponse(LangServerConstants.INTERNAL_ERROR_LINE,
-                        LangServerConstants.INTERNAL_ERROR, message, null);
-            }
+            TextDocumentPositionParams posParams = gson.fromJson(params.toString(), TextDocumentPositionParams.class);
+
+            Position position = posParams.getPosition();
+            String textContent = posParams.getText();
 
+            SuggestionsFilterDataModel filterDataModel = new SuggestionsFilterDataModel();
+            CapturePossibleTokenStrategy errStrategy = new CapturePossibleTokenStrategy(position, filterDataModel);
+
+            CompilerContext compilerContext = new CompilerContext();
+            compilerContext.put(DefaultErrorStrategy.class, errStrategy);
+
+            options = CompilerOptions.getInstance(compilerContext);
+            options.put(SOURCE_ROOT, "/home/nadeeshaan/Desktop");
+            options.put(COMPILER_PHASE, "typeCheck");
+
+            Compiler compiler = Compiler.getInstance(compilerContext);
+            // here we need to compile the whole package
+            BLangPackage bLangPackage = compiler.compile("test.bal");
```

Comment:
Can we avoid hardcoded file names?

Reply:
This will be replaced accordingly, with the client side request path is completed and this hardcoded path is temporarily added.